File: frontend/src/app/services/MedicalStaffService.ts
import {Injectable} from '@angular/core';
import {map} from 'rxjs/operators';
import {ApiService} from "./ApiService";
import {ConfigService} from "./ConfigService";
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { MedicalStaff } from '../model/Users';
import { Observable } from 'rxjs';
import { BloodBank } from '../model/BloodBank';

@Injectable({
  providedIn: 'root'
})
export class MedicalStaffService {

  headers: HttpHeaders = new HttpHeaders({ 'Content-Type': 'application/json' });

  constructor(
    private apiService: ApiService,
    private config: ConfigService,
    private http: HttpClient
  ) {
  }

  updateMedicalStaff (medicalStaff: MedicalStaff){
    return this.apiService.post(this.config.medical_staff_url, medicalStaff);
  }
  registerMedicalStaff (medicalStaff: MedicalStaff){
    return this.apiService.post(`${this.config.medical_staff_url}/register`, medicalStaff);
  }
  getBloodBank (): Observable<BloodBank>{
    return this.apiService.get(`${this.config.medical_staff_url}/bank`);
  }

}

File: backend/src/main/java/com/example/isa/service/interfaces/MedicalStaffService.java
package com.example.isa.service.interfaces;
import com.example.isa.model.MedicalStaff;
import com.example.isa.model.User;

import java.util.List;

import com.example.isa.model.BloodBank;
import com.example.isa.model.MedicalStaff;
import java.util.List;
import java.util.UUID;

public interface MedicalStaffService {
    List<MedicalStaff> getAll();
    MedicalStaff getById(UUID id);

    MedicalStaff updateMedicalStaff(MedicalStaff medicalStaff);
    boolean register(MedicalStaff medicalStaff);

    BloodBank getBloodBank(MedicalStaff medicalStaff);

}
